### DISCUSSION TITLE
Wireless RS-485 Adapter?

### DISCUSSION THREAD

**Discussion by tinkerer9** [aqualinkd/AqualinkD#271] [Q&A] [answered]
I already have a Raspberry Pi 4, and I don't want to buy a Raspberry Pi Zero. However, my Jandy Aqualink panel is relatively far from the Pi. Is it possible to use an RS-485 Bluetooth adapter like [this one](https://www.amazon.com/Bluetooth-Industrial-Wireless-Communication-Converter/dp/B0CFLM815P/ref=sr_1_1_sspa?crid=1I5R1Y36QP9RV&dib=eyJ2IjoiMSJ9.k2r7KuRzbQOWR4a7LnBMyD_wJVs7uimqOAts6FohWXmYmaWVLfO9CV0MVvytF6DH-zN6h9bdFYiKARKmMzin5WOfd9ID-2NMJPF-8YUBwHTd46Xp7x8_XVyTmFOjvb-hrfyt6rqeSywhXiTIgQVZyCSNkI8Adx-md_geyRlzf7C7LtMibBYqyyxJcYYpnxK5sN8brnOzOtxOU2lTxUz91V2TTyxgzUiY2J6tO6ItUlE.7qRQkRsW4YUajbTzXOWlnF7q-qn5ScZxytpyZYIhKvs&dib_tag=se&keywords=wireless%2Brs485&qid=1711931303&sprefix=wireless%2Brs485%2Caps%2C159&sr=8-1-spons&sp_csd=d2lkZ2V0TmFtZT1zcF9hdGY&th=1)?
Thanks!

**Comment by AUPoolUser** [comment]
I cannot answer your question - but I had a similar problem of having a pi far away from the aqualink panel.  So I ran a cat6 cable from my Aqualink Panel about 25 metres to my Pi3 which has the normal RS485 adapter plugged into the pi. I paired the colour pairs together thinking it would help reduce data loss - not sure if it did. But It’s been working since September 2018 without one issue.  I really appreciate the AqualinkD!

if your able to run a cat6 cable with ease I would recommend that.

**Comment by sfeakes** [comment, marked as answer]
I have never tried that device, but from the limited documentation it should work. Latency would be my main concern, but without trying I can’t say for sure.
BTW AqualinkD won’t work on a pico, it needs a Pi.  I think you may be confusing a Pi Zero with a Pico.

**Reply by tinkerer9** [reply]
Sorry, I meant Pi Zero. I'll try the adapter! Thank you!

**Comment by tinkerer9** [comment]
In case anyone is wondering the same thing as me:

I ended up running a cable (type doesn't really matter as long as it has **2 conductors**) to an [RS-485 to USB adapter](https://www.amazon.com/gp/product/B0BNZBNPWL/ref=ox_sc_act_title_2?smid=A3B0XDFTVR980O&th=1) which was plugged into my **Raspberry Pi 4B** running **Homebridge**.
I since switched to **Home Assistant** and have a **Raspberry Pi Zero 2 W** running AqualinkD, which connects to my Raspberry Pi running Home Assistant via MQTT&mdash;the instructions on the wiki are very helpful.

All of this works very well.
If you have wiring running through your house to a controller, consider piggybacking off of that.

The wiki also mentions a **RS-485** HAT that @sfeakes made that connects to the Jandy panel using all four wires so the Pi Zero uses its power. If you have the Pi Zero in the panel or somewhere nearby, you can just use Wi-Fi without running any cables (I haven't tried this out).

**Reply by rcork** [reply]
I bought a HAT from @sfeakes and it is great. However, the wifi on the RPi Zero is pretty weak because there's not external antenna and the Pi was in my RS-8 steel enclosure which kills wifi signal. It would lose connection to HA, almost impossible to connect to the Pi over SSH for any maintenance. Mosh made it better (highly recommend mosh for any terminal junkies like me that need to ssh over high latency network connections). Fortunately i already had ethernet running to my iAqualink that the aqualinkD replaced so once i connected the RPi/aqualinkD over ethernet, its been rock solid.

**Reply by eivancic77** [reply]
I'm a huge fan of the RS-485 HAT—the convenience of having it power the board directly is awesome. If anyone else is running into WiFi drops because the Pi is trapped inside that heavy steel RS enclosure, I highly recommend modifying your setup with an external antenna.

I've had great success with these specs:

- Antenna: Dual-Band Omni-directional (2.4GHz / 5GHz)
- Gain: 7dBi
- Mounting: Routed directly through the metal case using an M12 threaded base

Getting the antenna outside of the steel box completely eliminated my connectivity issues and made the connection rock solid.

**Reply by jkosharek** [reply]
@eivancic77  I second everything said. I got the Radxa Zero 3W with eMMC Storage and the 3D Printed case https://aqualinkd.github.io/purchase/ both components are very high quality. Been running for over a year flawlessly.

**Reply by eivancic77** [reply]
To add to this, if you are moving to an external antenna, there are two critical factors for success: using a proper ground plane and antenna placement/orientation.

Most omni-directional external antennas (like the 7dBi models mentioned above) are designed to use the metal enclosure itself as a "ground plane." When you mount the threaded base directly through the steel box, the metal casing acts as a reflector that shapes and stabilizes the signal. If you were to mount it to plastic or leave it ungrounded, the antenna's performance and range would drop significantly. For the best results, try to mount the antenna as close to the center of that metal surface as possible; putting it too close to an edge or side can unevenly "pull" and distort the signal pattern.

Additionally, orientation is key. "Omni-directional" means the signal radiates outward in a doughnut shape perpendicular to the antenna mast, not a perfect sphere. For the best coverage across your yard or back to your home router, make sure the antenna is mounted perfectly vertical and placed as high up or clear of immediate obstructions as possible.
